### ISSUE TITLE
[EDGE 2.3.7.0] gRPC sharing : Causing chef and mkdir failures

### ISSUE THREAD

**Issue by NorseGaud** [docker/for-mac#4943]
- [X] I have tried with the latest version of my channel (Stable or Edge)
  - [X] I have uploaded Diagnostics
  - Diagnostics ID: 6351D07E-AAED-4A65-ACAB-F28E1D68BFEA/20200925132901

### Expected behavior

It to not error out.

### Actual behavior

```
❯ ./GITLAB/install-gitlab-on-docker.bash                 
]] Cleaning up the previous GitLab installation
Error response from daemon: No such container: anka-gitlab-runner-shared
Error: No such container: anka-gitlab-runner-shared
Error response from daemon: No such container: anka-gitlab-runner-project-specific
Error: No such container: anka-gitlab-runner-project-specific
ERROR: 
        Can't find a suitable configuration file in this directory or any
        parent. Are you in the right directory?

        Supported filenames: docker-compose.yml, docker-compose.yaml
        
Password:
127.0.0.1 anka.gitlab
]] Starting the GitLab Docker container
Creating network "gitlab_default" with the default driver
Creating anka.gitlab ... error

ERROR: for anka.gitlab  Cannot start service anka.gitlab: error while creating mount source path '/Users/nathanpierce/anka.gitlab-data/config': mkdir /Users/nathanpierce/anka.gitlab-data: file exists

ERROR: for anka.gitlab  Cannot start service anka.gitlab: error while creating mount source path '/Users/nathanpierce/anka.gitlab-data/config': mkdir /Users/nathanpierce/anka.gitlab-data: file exists
ERROR: Encountered errors while bringing up the project.
```

### Information
  - macOS Version: 10.15.6

### Diagnostic logs
6351D07E-AAED-4A65-ACAB-F28E1D68BFEA/20200925132901

### Steps to reproduce the behavior
1. Make sure gRPC FUSE is enabled + using EDGE
2. Run https://github.com/veertuinc/getting-started/blob/master/GITLAB/install-gitlab-on-docker.bash
3. It may pass the first time, but start throwing Chef errors. Cancel it if so.
4. On the second run, you'll get the failure provided above.

The chef errors are related to the gRPC too. Turning gRPC off allows the script to run.

**Comment by stephen-turner** [comment]
Thanks for the report, @NorseGaud.
@djs55 PTAL.

**Comment by NorseGaud** [comment]
Unrelated: I think it's cool that I'm in Cambridge, MA (USA) and both you @stephen-turner and @djs55 are in Cambridge, UK :)

**Comment by stephen-turner** [comment]
It reminds me of one time when I saw a ranking of the top 100 universities in the world and all of the top 3 were in a Cambridge.

**Comment by stephen-turner** [comment]
But going back to the bug, @djs55 has now reproduced it, thank you for the clear repro case.

**Comment by djs55** [comment]
The fixes for the 2 issues seen here:

- a stale cache entry causing `mkdir: file exists`
- an assertion failure trying to set the `g+s` bit for a directory

have both been released as part of [stable 2.4.0.0](https://desktop.docker.com/mac/stable/48506/Docker.dmg). I was able to run the script and launch `gitlab` with this version.

I'll close this ticket but feel free to open fresh ones if you have further problems.

Thanks for your report and clear repro steps!

**Comment by simonferquel** [comment]
The issue has been fixed in Edge release 2.4.2.0, thank you for reporting it.
